projects/calc/js/main.js

```diff
@@ -0,0 +1,365 @@
+      this._result = this._calcResult({
+        firstOperator: this._operand2 || this._operand1,
+        operator: operator
+      });
+
+      if(!this._operator) {
+        this._operand1 = this._result
+      } else {
+        this._operand2 = this._result
+      }
+    }
+
+    // Complex math function
+    function complexMath() {
+      if(this._operator) {
+        this._result = this._calcResult({
+          firstOperator: this._operand1,
+          secondOperator: this._operand2 || this._operand1,
+          operator: operator
+        });
+
+        this._operand2 = this._result;
+      }
+    }
+  }
+
+  // Handle decimal point
+  _handlePoint(value) {
+    let activeOperand = this._activeOperand;
+
+    if(activeOperand.indexOf('.') === -1) {
+      if(activeOperand.charAt(0) === '') {
+        value = '0.';
+      }
+      this._handleOperand(value);
+    }
+  }
+
+  // Handle backspace button
+  _handleRemoveButton() {
+    this._result = this._activeOperand = this._activeOperand.slice(0, -1);
+  }
+
+  // Handle change sign button
+  _handleChangeSign() {
+    let activeOperand = this._activeOperand;
+
+    if(activeOperand) {
+      if(activeOperand.charAt(0) === '-') {
+        this._activeOperand = activeOperand.slice(1);
+      } else {
+        this._activeOperand = '-' + activeOperand;
+      }
+
+      this._result = this._activeOperand;
+    }
+  }
+
+  // Result button handler
+  _handleResult() {
+    if(this._operator) {
+      this._copyOperands();
+    }
+
+    this._interimResult();
+  }
+
+  // Clear button handler
+  _handleClear() {
+    this._operand1 = '';
+    this._operand2 = '';
+    this._operator = null;
+    this._activeOperand = '';
+    this._result = '';
+    this._hasError = false;
+  }
+
+  // Calculation result
+  _calcResult(obj) {
+    let result;
+
+    try {
+      result = this._allOperators[obj.operator]({
+        a: parseFloat(obj.firstOperator) || 0,
+        b: parseFloat(obj.secondOperator) || 0
+      });
+    } catch(e) {}
+
+    if(!isNumeric(result)) {
+      this._hasError = true;
+      return 'Недопустимое значение';
+    }
+
+    function isNumeric(n) {
+      return !isNaN(parseFloat(n)) && isFinite(n);
+    }
+
+    return +result.toFixed(20);
+  }
+
+  // Copy active operand
+  _copyOperands() {
+    if(this._activeOperand) {
+      if(!this._operand1) {
+        this._operand1 = this._activeOperand;
+      } else {
+        this._operand2 = this._activeOperand;
+      }
+
+      this._activeOperand = '';
+    }
+  }
+
+  // Show result
+  _showResult(value) {
+    this._resultField.value = value;
+  }
+
+  _interimResult() {
+    if(this._operand1 && this._operand2 && this._operator) {
+      this._result = this._calcResult({
+        firstOperator: this._operand1,
+        secondOperator: this._operand2,
+        operator: this._operator
+      });
+
+      this._operand1 = this._result;
+      this._operand2 = this._operator = '';
+    }
+  }
+}
+
+let calc = new Calc({
```

Comment:
in general this should be in separate file 
something like `app.js` or `main.js`